src/server/test-suites/transformers/test-suite-run-results.js

```diff
@@ -79,7 +79,14 @@ function transformTestSuiteRunResults(testRun) {
       value: testRun.user.displayName
     },
     { kind: 'text', value: getDuration(testRun, hasResult) },
-    { kind: 'date', value: testRun.taskLastUpdated }
+    { kind: 'date', value: testRun.taskLastUpdated },
+    runTaskStatus === taskStatus.inProgress
+      ? {
+          kind: 'button',
+          value: 'Stop',
+          url: `/test-suites/${testRun.testSuite}/${testRun.runId}/stop`
+        }
+      : { kind: 'text', value: '' }
```

Comment:
So its worth noting that when a value is falsey and not a number then the default `noValue` is provided. Have a look at the if/else clause here 

- https://github.com/DEFRA/cdp-portal-frontend/blob/main/src/server/common/components/entity/template.njk#L21
- https://github.com/DEFRA/cdp-portal-frontend/blob/main/src/server/common/components/entity/template.njk#L84

What that means is this could simply become:

```
{
  kind: 'button',
  value: runTaskStatus === taskStatus.inProgress ? 'Stop' : '',
  url: `/test-suites/${testRun.testSuite}/${testRun.runId}/stop`
}

```

Reply:
Or maybe even a disabled button when the test is not running?

```
{
  kind: 'button',
  value: 'Stop',
  url: `/test-suites/${testRun.testSuite}/${testRun.runId}/stop`,
  attributes: runTaskStatus === taskStatus.inProgress ?  {}: {
    disabled: 'disabled"
  } 
}
```

`src/server/common/components/entity/template.njk`

```
...
    {% elseif params.kind === "button" %}
      <form action="{{ params.url }}" method="POST">
        <input type="hidden" name="csrfToken" value={{csrfToken}}>
        {{ govukButton({
             classes: "app-button app-button--secondary",
             text: params.value,
             attributes: params.attributes | default({})
           })
        }}
      </form>
...
```